src/main/java/pl/javastart/task/UniversityApp.java

```diff
@@ -12,111 +23,268 @@ public class UniversityApp {
+         * W przypadku gdy prowadzący ze wskazanym id nie istnieje wyświetla się komunikat:
+         * "Prowadzący o id [id prowadzacego] nie istnieje"
+         *
+         * @param code       - unikalny kod grupy
+         * @param name       - nazwa przedmiotu (np. "Podstawy programowania")
+         * @param lecturerId - identyfikator prowadzącego. Musi zostać wcześniej utworzony za pomocą metody {@link #createLecturer(int, String, String, String)}
+         */
     public void createGroup(String code, String name, int lecturerId) {
+        if (groupExists(code)) {
+            System.out.println("Grupa " + code + " juz istnieje.");
+        } else if (!lecturerExists(lecturerId)) {
+            System.out.println("Prowadzacy o id " + lecturerId + " nie istnieje.");
+        } else {
+            Lecturer lecturer = findLectuer(lecturerId);
+            Group group = new Group(code, name, lecturer);
+            groups[groupsCounter] = group;
+            groupsCounter++;
+        }
+    }
 
+    private Lecturer findLectuer(int lecturerId) {
+        int indexLecturer = 0;
+        for (int i = 0; i < lecturersCount; i++) {
+            if (lecturers[i].getId() == lecturerId) {
+                indexLecturer = i;
+            }
+        }
+        return lecturers[indexLecturer];
     }
 
+    private boolean groupExists(String code) {
+        boolean groupCheck = false;
+        for (int i = 0; i < groupsCounter; i++) {
+            if (groups[i].getCode().equals(code)) {
+                groupCheck = true;
+            }
+        }
+        return groupCheck;
+    }
 
     /**
-     * Dodaje studenta do grupy zajęciowej.
-     * W przypadku gdy grupa zajęciowa nie istnieje wyświetlany jest komunikat:
-     * "Grupa [kod grupy] nie istnieje
-     *
-     * @param index     - unikalny numer indeksu studenta
-     * @param groupCode - kod grupy utworzonej wcześniej za pomocą {@link #createGroup(String, String, int)}
-     * @param firstName - imię studenta
-     * @param lastName  - nazwisko studenta
-     */
+         * Dodaje studenta do grupy zajęciowej.
+         * W przypadku gdy grupa zajęciowa nie istnieje wyświetlany jest komunikat:
+         * "Grupa [kod grupy] nie istnieje
+         *
+         * @param index     - unikalny numer indeksu studenta
+         * @param groupCode - kod grupy utworzonej wcześniej za pomocą {@link #createGroup(String, String, int)}
+         * @param firstName - imię studenta
+         * @param lastName  - nazwisko studenta
+         */
     public void addStudentToGroup(int index, String groupCode, String firstName, String lastName) {
-
+        if (!groupExists(groupCode)) {
+            System.out.println("Grupa " + groupCode + " nie istnieje.");
+        } else {
+            int indexGroup = findGroup(groupCode);
+            int counter = groups[indexGroup].getStudentsCounter();
+            Student student = new Student(index, firstName, lastName);
+            if (!studentExists(index)) {
```

Comment:
Tutaj przydałaby się metoda wyszukująca studenta. Jeżeli student istnieje w tablicy students, to powinnaś go pobrać i tylko dopisać do grupy
```
Student student = findStudent(index);
if (student == null) {
 ..utwórz studenta
} 
... wstaw studenta do grupy
```